credentials/tls/certprovider/spiffe/spiffe_test.go

```diff
@@ -0,0 +1,136 @@
+/*
+ *
+ * Copyright 2025 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package spiffe
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/spiffe/go-spiffe/v2/bundle/spiffebundle"
+	"google.golang.org/grpc/testdata"
+)
+
+var SpiffeTrustBundle map[string]*spiffebundle.Bundle
+
+func TestKnownSpiffeBundle(t *testing.T) {
+	bundles, err := LoadSpiffeBundleMap(testdata.Path("spiffe/spiffebundle.json"))
+	if err != nil {
+		t.Fatalf("Error during parsing: %v", err)
+	}
+	if len(bundles) != 2 {
+		t.Fatal("did not parse correct bundle length")
+	}
+	if bundles["example.com"] == nil {
+		t.Fatal("expected bundle for example.com")
+	}
+	if bundles["test.example.com"] == nil {
+		t.Fatal("expected bundle for test.example.com")
+	}
+
+	expectedExampleComCert := loadX509Cert(testdata.Path("spiffe/spiffe_cert.pem"))
+	expectedTestExampleComCert := loadX509Cert(testdata.Path("spiffe/server1_spiffe.pem"))
+	if !bundles["example.com"].X509Authorities()[0].Equal(expectedExampleComCert) {
+		t.Fatalf("expected cert for example.com bundle not correct.")
+	}
+	if !bundles["test.example.com"].X509Authorities()[0].Equal(expectedTestExampleComCert) {
+		t.Fatalf("expected cert for test.example.com bundle not correct.")
+	}
+
+}
+
+func loadX509Cert(filePath string) *x509.Certificate {
+	certFile, _ := os.Open(filePath)
+	certRaw, _ := io.ReadAll(certFile)
+	block, _ := pem.Decode([]byte(certRaw))
+	if block == nil {
+		panic("failed to parse certificate PEM")
+	}
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		panic("failed to parse certificate: " + err.Error())
```

Comment:
Instead of calling `panic`, please pass `t.Testing` to this function and call `t.Fatalf` here. Also mark the function as a test helper. See: go/go-style/decisions#mark-test-helpers